test/tools/jsontests/LegacyTestsBoost.cpp

```diff
@@ -0,0 +1,221 @@
+#include <test/tools/jsontests/BlockChainTests.h>
+#include <test/tools/jsontests/StateTests.h>
+#include <test/tools/libtesteth/TestHelper.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace dev::test;
+namespace fs = boost::filesystem;
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTests";
+}
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTestsFiller";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTests/GeneralStateTests";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTestsFiller/GeneralStateTests";
+}
+
+template <class T>
+class LegacyTestFixtureBase
+{
+public:
+    LegacyTestFixtureBase(bool _onlyRunIfOptionsAllSet)
+    {
+        T suite;
+        if (Options::get().fillchain || Options::get().filltests)
+            BOOST_FAIL("Legacy tests are sealed and not refillable!");
+
+        string casename = boost::unit_test::framework::current_test_case().p_name;
+        boost::filesystem::path suiteFillerPath = suite.getFullPathFiller(casename).parent_path();
+        static vector<string> const timeConsumingTestSuites{
+            string{"stTimeConsuming"}, string{"stQuadraticComplexityTest"}};
+
+        bool skipTheTest = false;
+        if (test::inArray(timeConsumingTestSuites, casename) && !test::Options::get().all)
+            skipTheTest = true;
+        if (_onlyRunIfOptionsAllSet && !Options::get().all)
+            skipTheTest = true;
+
+        if (skipTheTest)
+        {
+            std::cout << "Skipping " << casename << " because --all option is not specified.\n";
+            test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
+            return;
+        }
+        suite.runAllTestsInFolder(casename);
```

Comment:
this is just a rant, we won't change it now, but this should have been called from the test body instead of weirdly from fixture constructor; for that you could have created a macro wrapping `BOOST_AUTO_TEST_CASE` macro always calling it from the body
(maybe do that in retesteth if it's any similar there)

Reply:
Please explain better the idea.

Reply:
Don't run the tests from the constructor, run them from the test body.
```
#define RETESTETH_AUTO_TEST_CASE(casename) BOOST_AUTO_TEST_CASE(casename) { m_suite.runAllTestsInFolder(casename); }
```

...anyway it's better to get rid of boost.test altogether

Reply:
I think it will get more confusing if to continue refactoring it that way